pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -816,7 +816,7 @@ def multi_head_attention_forward(
 
 
 def cosine_similarity(x1: Tensor, x2: Tensor, dim: int=1, eps: float=1e-8) -> Tensor:
-  numerator = modules.Dot()(x1, x2, reduce_dim_a=dim, reduce_dim_b=dim)
+  numerator = (x1 * x2).sum(dim=dim)
```

Comment:
Why did you change this? I think it's more efficient to use the `Dot` module. Unless you don't care about this here?

Reply:
As I commented above, this is the only use of the `Dot` module. So if it's possible to rewrite and do it without, I thought that'd be cleaner.